### ISSUE TITLE
Duplicate filed names when renaming fields is required

### ISSUE THREAD

**Issue by Threated** [oxidecomputer/typify#384]
When a type has 2 fields one is named `url` and the other is named `_url` this crate will produce a struct with 2 fields named `url` although a filed named `_url` would be valid.

**Comment by ahl** [comment]
Yeah. That sucks. It's related to #1 (the first issue i.e. the biggest hairball) unique naming of all things is a pain in the neck.

In your case, it sounds like you'd want fields named `_url` and `url`. As you know, the leading underscore has meaning in Rust to suppress unused / dead code warnings. It seemed undesirable to preserve the name and stumble into those semantics. What do you think?

**Comment by Threated** [comment]
Yeah I can't really think of a perfect solution for that either. Maybe there could be a to provide a mapping Funktion that maps invalid chars to a string or something. This way the user would be able to decide how to handle these fields an would not see surprising field names.
